### ISSUE TITLE
args[module]: missing required arguments: required_param1, required_param2 (warning)

### ISSUE THREAD

**Issue by OscarBell** [ansible/ansible-lint#3199]
##### Summary

The args module doesn't consider arguments specified via the args task keyword and thus fails the check when some or all of the required arguments are part of the args keyword.

##### Issue Type

* bug report

##### OS / ENVIRONMENT

```console
ansible-lint 6.14.2 using ansible 2.14.3
```

- ansible installation method: pip,
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

An example module that has 3 required parameters (required_param{1,2,3}) and 2 optional parameters (optional_params{1,2}) are covered correctly by ansible-lint when specifying the parameter directly as module arguments in the task like below.

```yaml
- name: Ansible-lint for args rule should succeed
  demo_module:
    required_param1: required_param1
    required_param2: required_param2
    required_param3: required_param3
    optional_param1: true
```

If some of the parameters are specified via the task args keyword ansible-lint doesn't collect these arguments and thus results in a warning for the args rule.

```yaml
    - name: Ansible-lint for args rule should succeed
      demo_module:
        required_param3: required_param3
        optional_param1: true
      args:
        demo_module_vars:
          required_param1: required_param1
          required_param2: required_param2

## Alternative test case which also results in a warning

    - name: Set demo_module_vars
       ansible.builtin.set_fact:
         demo_module_vars:
           required_param1: required_param1
           required_param2: required_param2


    - name: Ansible-lint for args rule should succeed
      demo_module:
        required_param3: required_param3
        optional_param1: true
      args: "{{ demo_module_vars }}"
```

Arguments for the args keyword can be provided directly as a dict or via a variable that is defined elsewhere in the playbook.

The module_args for the demo module are specified as below and confirm the validation ansible-lint is warning about.

```python
module_args = dict(
    required_param1=dict(type='str', required=True),
    required_param2=dict(type='str', required=True),
    required_param3=dict(type='str', required=True),
    optional_param1=dict(type='bool', required=False, default=False),
    optional_param2=dict(type='bool', required=False, default=False),
)
```

Ansible-lint output

```console
$ ansible-lint playbook.yml -p
WARNING  Listing 1 violation(s) that are fatal
playbook.yml:24: args[module]: missing required arguments: required_param1, required_param2 (warning)
Read documentation for instructions on how to ignore specific rule violations.

                  Rule Violation Summary                   
 count tag          profile rule associated tags           
     1 args[module]         syntax, experimental (warning) 

Passed with production profile, 5/5 star rating: 0 failure(s), 1 warning(s) on 1 files.
```

##### Desired Behavior

Expect ansible-lint to not fail the rule because all required arguments are specified in the task and a normal ansible playbook run doesn't have any missing required arguments.

##### Actual Behavior

```console
$ ansible-lint playbook.yml -p
WARNING  Listing 1 violation(s) that are fatal
playbook.yml:24: args[module]: missing required arguments: required_param1, required_param2 (warning)
Read documentation for instructions on how to ignore specific rule violations.

                  Rule Violation Summary                   
 count tag          profile rule associated tags           
     1 args[module]         syntax, experimental (warning) 

Passed with production profile, 5/5 star rating: 0 failure(s), 1 warning(s) on 1 files.
```

I'm happy to assist in solving this issue.

**Comment by OscarBell** [comment]
The first case can be solved but the second case probably not because it contains a Jinja template.

```python
from ansible.parsing.yaml.objects import AnsibleMapping
# Around line 111 in rules/args.py
        for key in [_k for _k in module_args.keys() if isinstance(module_args[_k], AnsibleMapping)]:
            module_args.update({
                key: value
                for key, value in module_args[key].items()
                if not key.startswith('__')
            })
            del module_args[key]
```

When specifying
```yaml
args: "{{ demo_module_vars }}"
```
the `module_args["_variable_params"]` contain the value `{{ demo_module_vars }}` which ansible-lint probably can't translate.

**Comment by ssbarnea** [comment]
Can you please provide an example from builtin or popular collections galaxy where this bug occurs instead of a custom module? If not, we will close this as wontfix.

**Comment by OscarBell** [comment]
@ssbarnea An example, with for instance the builtin file module, fails as well when using ansible-lint 6.14.2 but I noticed that ansible-lint 6.16 fixed the issue for both cases (demo_module and builtin.file). Thanks for fixing the issue in #3402.

Also noticed that one of my test cases was not correct, the one which provided a nested dict for the args argument. That will never work for the module.

```yaml
  vars:
    file_path_vars:
      path: foo.conf
  tasks:
    - name: Change file ownership, group and permissions
      ansible.builtin.file:
        # path: foo.conf
        owner: foo
        group: foo
        mode: '0644'
      args: "{{ file_path_vars }}"
```

Anslble-lint warning
```yaml
args[module]: missing required arguments: path (warning)
playbook.yml:47 Task/Handler: Change file ownership, group and permissions
```

**Comment by OscarBell** [comment]
Closed because #3402 fixed this issue as well.
